homeassistant/components/flexpool/sensor.py

```diff
@@ -0,0 +1,74 @@
+"""Support for Etherscan sensors."""
+from datetime import timedelta
+
+import flexpoolapi
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_ADDRESS, CONF_DEVICES, CONF_SERVICE_DATA
+import homeassistant.helpers.config_validation as cv
+
+from .sensors import (
+    FlexpoolBalanceSensor,
+    FlexpoolHashrateSensor,
+    FlexpoolPoolHashrateSensor,
+    FlexpoolPoolLuckSensor,
+    FlexpoolPoolWorkersSensor,
+    FlexpoolWorkerHashrateSensor,
+    FlexpoolWorkerShareSensor,
+)
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_ADDRESS): cv.string,
+        vol.Optional(CONF_DEVICES, default=False): cv.boolean,
+        vol.Optional(CONF_SERVICE_DATA, default=False): cv.boolean,
+    }
+)
```

Comment:
We no longer allow integrations to add or change a platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Please note that this integration connects to a device or service and another Architecture Decision Record applies that disallows the use of YAML configuration in favor of a configuration flow via the UI:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

As these changes often involve a bit of work and some significant shift in the current code, we will close this PR for now.

We (and our community!) would really appreciate it if you could start on the refactoring of this integration in a new PR.

Thanks already! :+1: